library/Makefile

```diff
@@ -1,8 +1,21 @@
 
 # Also see "include/mbedtls/config.h"
 
+ifeq ($(CC),iccarm)
+CFLAGS	?= --arm -Om --silent
+
+# gcc comparable warning flags are not present in IAR.
+# All warnings are enabled by default.
+WARNING_CFLAGS ?=
+else
 CFLAGS	?= -O2
 WARNING_CFLAGS ?=  -Wall -W -Wdeclaration-after-statement
+endif
+
+# IAR archive command does not have ar compatible options 
+ifneq ($(AR),iarchive)
```

Comment:
I think this is a very fragile way of detecting whether we are compiling or archiving with IAR. For example, if I execute this makefile with the command `CC=\path\to\iccarm AR=\path\to\iarchive make lib` I do not think these conditional statements will take the expected branches for IAR causing a failure.